### ISSUE TITLE
Controller rotation offset

### ISSUE THREAD

**Issue by ArnbrandHelgi** [LivingFray/HaloCEVR#15]
I can't set the controller rotation offset in the settings file. Instead of the angle offset, controller starts to turn faster and in the wrong directions. Values 0.*** have no visible effect. Larger values ​​lead to the above results. 

```//[Vector3] Rotation added to the controller when calculating the in game hand rotation (Default Value: "(0, 0, 0)") ControllerRotation = (0, 0, 0);```

**Comment by benjamincheng10** [comment]
yeah the guns are tilted up too high, about 45 degrees. Would like to be able to change that without the rotation becoming all wonky.

**Comment by Jazza159** [comment]
I'm seeing this too with the Vive controllers.

**Comment by LivingFray** [comment]
Controller aiming has been the bane of my existence working on this mod. I've tried to account for it by using the controller's "tip pose" to get the forward vector but that's apparently not enough.
Fixing this shouldn't require any halo specific knowledge/reverse engineering, so anyone who's willing is free to try fix this and submit a PR.

**Comment by DrMirakle** [comment]
What's interesting is that when using the gun with two hands, it's possible to rotate the gun to be in the right rotation for the main hand, you'd think it would just be a matter of dumping the gun position with two hands and with one hand and then add a rotation matrix that accounts for the difference (and make the rotation angle a setup variable).

**Comment by slipperfish** [comment]
I'm not a programmer but I tried to correct the pitch for my quest 2 and I think I realize what's going on. The config adjustment changes the pitch on the world axis instead of the local axis, or whatever you would call them.

![pitch](https://github.com/user-attachments/assets/b0cb8f0b-c0d7-4122-a36f-1da0c5dee181)

Sat in a swivel chair and punched my arms out with straight elbows, but the effect on aiming was the same whether I spun in my chair or turned with the analog stick. The config adjustment does work, but only in one direction. In the opposite direction it will have the opposite effect and pitch up. Aiming is mostly normal in these two directions. but If you are facing the medians between these directions, your gun will be rolled to the side and tracks diagonally. And all this fits a disk/sphere tilted on its side.

Anyway on controllers the pitch of the cursor vector is probably chosen on what's comfortable with lowered arms and bent elbows at ease. It's probably not a good indication of where the gun barrel should point when using a shooting pose. Thanks and I hope all this helps.

**Comment by slipperfish** [comment]
At first I thought the gun was pitched too high, but then I remembered gun handles aren't designed for your wrists in their natural position. In real life you're supposed to crank your wrists down and not "limp wrist". If I do that the defaults are good and I've gotten used to it. But it's trading comfort for realism, so that makes it really subjective as to what the angle should be. Halo isn't a mil-sim, but is in virtual reality now.

I also realized you can see your controllers overlayed on the game when you use the pause menu. 

Anyway this is just a lot of words just to say I don't really question the defaults on oculus anymore.

**Comment by DrMirakle** [comment]
For sure, even though the angle seems a bit exaggerated, I can get used to it, it's not wildly off (using a PSVR2 controller).
The problem is that when two-handing a weapon, the dominant hand controller has to be tilted much more vertical than when one-handing the guns, and this difference makes it difficult to switch back to one handing (it takes me always like 5 seconds to get re-used to the new angle).

**Comment by slipperfish** [comment]
> The problem is that when two-handing a weapon, the dominant hand controller has to be tilted much more vertical than when one-handing the guns

Can you re-describe what you mean? It's also an apples to oranges comparison because one handed aiming is a vector drawn from a single point from one controller with adjustments to how a theoretical gun would point from your hand, but two handed weapons are a vector drawn between two controllers, and it actually ignores the pitch tilt from your dominant hand, that is why I don't get your description. It uses roll from your dominant hand, do you mean roll?

**Comment by AirSpiritFox** [comment]
Oh lordy please let me know when someone has fixed the vives angle problems!  I tried what has already been suggested.  Want to get stuck in so badly to halo

**Comment by slipperfish** [comment]
```
//[Vector3] Rotation added to the controller when calculating the in game hand rotation (Default Value: "(0, 0, 0)")
ControllerRotation = (0, 0, 0)
```
I realized reading this again I might be the one confused about what this was supposed to fix. I thought it adjusted the game hand / weapon rotation, but I realize "rotation added to the controller when calculating" might mean it adjusts the controller's polled rotation. So if your controller's rotation is correctly reported, this will only break it.

> I also realized you can see your controllers overlayed on the game when you use the pause menu.

And now I'm realizing the weapon rotation shown in the pause menu and what is used in game are different. This fooled me into thinking my config needed adjustment. But it seems intentional, I don't know. But unless there's some explanation I suggest changing it.
